### PR TITLE
feat: report/alert list CRUD view

### PR DESCRIPTION
**Pull request by lilykuang**
### SUMMARY

- implemented list view for Alert and Report use same endpoint `/api/v1/report`
- alerts and reports determined by `type`
- add new antd component `Switch` and it to storybook
 - UI feature is behind feature flag `SIP_34_ALERTS_UI `
### BEFORE/AFTER SCREENSHOTS OR ANIMATED GIF

Screen Shot 2020-11-24 at 9 27 58 AM
Screen Shot 2020-11-24 at 9 27 21 AM
Screen Shot 2020-11-24 at 9 56 32 AM

### TEST PLAN

- unit test for alert/report list view

### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [x] Changes UI
- [ ] Requires DB Migration.
- [ ] Confirm DB Migration upgrade and downgrade tested.
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

File: superset-frontend/src/views/CRUD/alert/AlertList.tsx [hunk 1]
```diff
@@ -0,0 +1,303 @@
...
+          recipients.map((r: any) => (
+            <Icon key={r.id} name={r.type as IconName} />
+          )),
+        accessor: 'recipients',
+        Header: t('Notification Method'),
+        disableSortBy: true,
+      },
+      {
+        Header: t('Schedule'),
+        accessor: 'crontab',
```
Comment (lilykuang) [inline superset-frontend/src/views/CRUD/alert/AlertList.tsx:171]: This column will be updated later and keeping it as plain text column for now

File: superset-frontend/src/views/CRUD/alert/AlertList.tsx [hunk 2]
```diff
@@ -0,0 +1,303 @@
...
+  addDangerToast: (msg: string) => void;
+  addSuccessToast: (msg: string) => void;
+  isReportEnabled: boolean;
+}
+
+const StatusIcon = styled(Icon)<{ status: string }>`
+  color: ${({ status, theme }) => {
+    if (status === 'alerting') return '#FBC700';
+    if (status === 'failed') return theme.colors.error.base;
+    if (status === 'ok') return theme.colors.success.base;
```
Comment (villebro) [inline superset-frontend/src/views/CRUD/alert/AlertList.tsx:51]: Could we use `enum` for status here, potentially with a switch?

**Review by villebro — APPROVED** [review]
LGTM with one small typing proposal.

File: superset-frontend/src/views/CRUD/hooks.ts [hunk 3]
```diff
@@ -40,6 +41,7 @@ export function useListViewResource<D extends object = any>(
   handleErrorMsg: (errorMsg: string) => void,
   infoEnable = true,
   defaultCollectionValue: D[] = [],
+  baseFilters: FilterValue[] = [], // must memolize otherwise infinite loop
```
Comment (nytai) [inline superset-frontend/src/views/CRUD/hooks.ts:44]:
```suggestion
  baseFilters: FilterValue[] = [], // must be memoized
```

File: superset/app.py [hunk 4]
```diff
@@ -418,8 +419,10 @@ def init_views(self) -> None:
                 category_label=__("Manage"),
                 icon="fa-exclamation-triangle",
             )
-            appbuilder.add_view_no_menu(AlertObservationModelView)
             appbuilder.add_view_no_menu(AlertLogModelView)
+            appbuilder.add_view_no_menu(AlertObservationModelView)
+            if feature_flag_manager.is_feature_enabled("SIP_34_QUERY_SEARCH_UI"):
+                appbuilder.add_view_no_menu(AlertReportModelView)
```
Comment (nytai) [inline superset/app.py:425]: Is this the correct feature flag?

Comment (lilykuang) [reply superset/app.py:425]: ops! nice catch

**Review by riahk — APPROVED** [review]

**Review by nytai — APPROVED** [review]
lgtm. needs a rebase before merge though

**Pull request merged**